src/main/java/com/pusher/client/channel/ChannelEventListener.java

```diff
@@ -39,6 +39,28 @@
      *
      * @param channelName
      *            The name of the channel that was successfully subscribed.
+     * @param historyComplete
+     *            If true, the requested history was available in full.
+     *            If false, the requested history was not fully available, but
+     *            the subcription has been made. available history will be delivered
+     *            to the appropriate {@link SubscriptionEventListener#onEvent}
+     *            callbacks.
+     *            If null, history was not requested for this subscription.
      */
-    void onSubscriptionSucceeded(String channelName);
+    void onSubscriptionSucceeded(String channelName, Boolean historyComplete);
```

Comment:
We've generally avoided the term "history" so maybe we should also do that here.

That aside, `resumeSuccessful` may be more relevant/symmetrical to `resumeAfterId`.